Explore: Fix queries (cached & non) count in usage insights

Elfo404 opened the pull request:
**What is this feature?**

Fixes Explore query stats reporting for usage insights by reusing the `enrichWithDashboardInfo` function (renamed to `enrichWithInfo`).

Given that `enrichWithDashboardInfo` is only attaching dashboard info when there's an open dashboard the code is slightly simplified, and no special-casing for explore is needed.


**Why do we need this feature?**

Currently running queries in Explore reuslts in the total number of series reported as number of queries. data from usage-insights is therefore skewed.

**Which issue(s) does this PR fix?**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #https://github.com/grafana/support-escalations/issues/7863

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

Elfo404 committed:
Fix: Fix queries (cached & non) count in usage insights

Elfo404 commented on public/app/features/query/state/queryAnalytics.ts:
```diff
@@ -77,8 +66,8 @@ export function emitDataRequestEvent(datasource: DataSourceApi) {
 
-    if (data.error) {
-      eventData.error = data.error.message;
+    if (data.errors?.length) {
```
`data.error` is deprecated

Elfo404 committed:
also keep deprecated error property

axelavargas reviewed (approved):
Code looks good to me 😀, thanks for helping with this @Elfo404 🙌

Elfo404 committed:
Merge branch 'main' into gio/fix/explore-insights

Elfo404 committed:
Fix & refactor tests

Elfo404 commented:
@axelavargas i've fixed the code, at the end of the day we actually need some special casing for explore. i've also refactored a bit the tests to be a bit ore maintanable and covered the Correlations usecase that wasn't covered in tests (also removing a type assertion)

mind having another look please?

axelavargas reviewed (approved):
@Elfo404 I took another look, codewise looks good to me. Good catch about the errors. 

🏅 For the new tests!
